lib/core/versions/latest/AnchoredDataSerializer.ts

```diff
@@ -0,0 +1,95 @@
+import AnchoredData from './models/AnchoredData';
+import Encoder from './Encoder';
+import ErrorCode from './ErrorCode';
+import { SidetreeError } from '../../Error';
+
+/**
+ * Encapsulates functionality to serialize/deserialize data that read/write to
+ * the blockchain.
+ */
+export default class AnchoredDataSerializer {
+
+  private static readonly delimeter = '.';
+  private static readonly maxInThreeBytes = 0xFFFFFF;
+
+  /**
+   * Converts the given inputs to the string that is to be written to the blockchain.
+   *
+   * @param anchorFileAddress The anchor file address to write.
+   * @param numberOfOperations The number of operations to write.
+   */
+  public static serialize (dataToBeAnchored: AnchoredData): string {
+
+    // First convert the number of operations input into a 3-byte buffer and then base64 encode it
+    const numberAsBuffer = AnchoredDataSerializer.convertToThreeBytesBuffer(dataToBeAnchored.numberOfOperations);
+    const encodedNumberOfOperations = Encoder.encode(numberAsBuffer);
+
+    // Concatenate the inputs w/ the delimeter and return
+    return `${encodedNumberOfOperations}${AnchoredDataSerializer.delimeter}${dataToBeAnchored.anchorFileHash}`;
+  }
+
+  /**
+   * Deserializes the given string that is read from the blockchain into data.
+   *
+   * @param serializedData The data to be deserialized.
+   */
+  public static deserialize (serializedData: string): AnchoredData {
+
+    const splitData = serializedData.split(AnchoredDataSerializer.delimeter);
+
+    if (splitData.length !== 2) {
+      throw new SidetreeError(ErrorCode.AnchoredDataIncorrectFormat, `Input is not in correct format: ${serializedData}`);
+    }
+
+    const decodedNumberOfOperations = Encoder.decodeAsBuffer(splitData[0]);
+    const numberOfOperations = AnchoredDataSerializer.convertFromThreeBytesBuffer(decodedNumberOfOperations);
+
+    return {
+      anchorFileHash: splitData[1],
+      numberOfOperations: numberOfOperations
+    };
+  }
+
+  private static convertToThreeBytesBuffer (numberOfOperations: number): Buffer {
+
+    if (numberOfOperations < 0) {
+      // We are going to write the number as unint32

```

Comment:
Consider removing this line. I think it is misleading as we will write as uint24 (if there is such a thing).